thirdeye-server/src/main/java/ai/startree/thirdeye/core/AppAnalyticsService.java

```diff
@@ -115,51 +121,55 @@ private MonitoredMetricWrapper wrapMonitoredMetric(final AlertMetadataDTO metada
-        case NO_FEEDBACK:
-          matrix.incUnclassified();
-          break;
-        case NOT_ANOMALY:
-          matrix.incFalsePositive();
-          break;
-        case ANOMALY:
-        case ANOMALY_EXPECTED:
-        case ANOMALY_NEW_TREND:
-          matrix.incTruePositive();
-          break;
-        default:
-          log.error("Unsupported feedback type: {}", type);
-      }
+    // filter to get anomalies without feedback
+    final Predicate predicate = Predicate.EQ("anomalyFeedbackId", 0L);
+    if (filter != null) {
+      filter.setPredicate(Predicate.AND(filter.getPredicate(), predicate));
+    } else {
+      filter = new DaoFilter().setPredicate(predicate);
     }
+    matrix.addUnclassified((int) anomalyManager.countParentAnomalies(filter));
+    final Map<AnomalyFeedbackType, Long> typeMap = aggregateFeedbackTypes(anomalyFeedbacksSupplier.get());
+    matrix.addUnclassified(Math.toIntExact(typeMap.get(NO_FEEDBACK)));
+    matrix.addFalsePositive(Math.toIntExact(typeMap.get(NOT_ANOMALY)));
+    matrix.addTruePositive(Math.toIntExact(typeMap.get(ANOMALY))
+        + Math.toIntExact(typeMap.get(ANOMALY_EXPECTED))
+        + Math.toIntExact(typeMap.get(ANOMALY_NEW_TREND)));
     return matrix;
   }
 
-  private List<AnomalyFeedback> getAnomalyFeedbacks() {
-    return anomalyManager.findParentAnomaliesWithFeedback().stream()
+  private List<AnomalyFeedback> getAllAnomalyFeedbacks() {
+    return getAnomalyFeedbacks(null);
+  }
+
+  private List<AnomalyFeedback> getAnomalyFeedbacks(final DaoFilter filter) {
+    return anomalyManager.findParentAnomaliesWithFeedback(filter).stream()
         .map(MergedAnomalyResultDTO::getFeedback)
         .collect(Collectors.toList());
   }
 
-  public AnomalyStatsApi computeAnomalyStats() {
-    List<AnomalyFeedback> feedbacks = anomalyFeedbacksSupplier.get();
-    Map<AnomalyFeedbackType, Long> feedbackStats = new HashMap<>();
-    for(AnomalyFeedbackType type : AnomalyFeedbackType.values()) {
+  public AnomalyStatsApi computeAnomalyStats(final DaoFilter filter) {
+    final List<AnomalyFeedback> allFeedbacks = filter == null
+        ? anomalyFeedbacksSupplier.get()
+        : getAnomalyFeedbacks(filter);
+    return new AnomalyStatsApi()
+        .setTotalCount(anomalyManager.countParentAnomalies(filter))
+        .setCountWithFeedback((long) allFeedbacks.size())
+        .setFeedbackStats(aggregateFeedbackTypes(allFeedbacks));
+  }
+
+  private Map<AnomalyFeedbackType, Long> aggregateFeedbackTypes(final List<AnomalyFeedback> feedbacks) {
+    final Map<AnomalyFeedbackType, Long> feedbackStats = new HashMap<>();
+    for(final AnomalyFeedbackType type : AnomalyFeedbackType.values()) {
       feedbackStats.put(type, 0L);
     }
     feedbacks.forEach(feedback -> {
-      AnomalyFeedbackType type = feedback.getFeedbackType();
-      long count = feedbackStats.get(type);
+      final AnomalyFeedbackType type = feedback.getFeedbackType();
+      final long count = feedbackStats.get(type);
```

Comment:
Minor (only a suggestio,)
These days I tend to prefer
```
feedbackStats.computeIfAbsent(type, () -> 0)
```

it allows to remove
```
    for(final AnomalyFeedbackType type : AnomalyFeedbackType.values()) {
      feedbackStats.put(type, 0L);
    }
```

Reply:
The for loop also handles initialising each feedback type to 0 so that the missing feedback types will show the 0 count to put out all the possible feedback types with their counts.

Reply:
oh good point